core/src/scheduler/task/SearchTask.cpp

```diff
@@ -159,8 +159,16 @@ XSearchTask::Load(LoadType type, uint8_t device_id) {
             stat = index_engine_->CopyToFpga();
             type_str = "CPU2FPGA";
         } else if (type == LoadType::CPU2APU) {
-            stat = index_engine_->CopyToApu(file_->row_count_);
-            type_str = "CPU2APU";
+            if (auto job = job_.lock()) {
+                auto search_job = std::static_pointer_cast<scheduler::SearchJob>(job);
+                if (search_job->index_files().size() > 1) {
```

Comment:
Why here only allow one segment file to search?

Reply:
Hi , 

Loading data to the Apu is expensive action. In order to load the data only once we need all the search data in one place so we will load it all together once . 
since tasks is been created acording to the number of segment files and each segment file contain part of the dataset,
we have to make sure all the data is been farward to APU in one action.